src/Uno.UI.BindingHelper.Android/Uno/UI/UnoViewGroup.java

```diff
@@ -223,9 +225,37 @@ public final void setNativeTextBlockLayout(android.text.Layout layout, int leftP
 		_topTextBlockPadding = topPadding;
 	}
 
+	@Override
+	protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec)
+	{
+		super.onMeasure(widthMeasureSpec, heightMeasureSpec);
+		_measureRequested = false;
+	}
+
 	@Override
 	protected void onDraw(android.graphics.Canvas canvas)
 	{
+		if(_measureRequested && !isLayoutRequested())
+		{
+			// Sometimes the isLayoutRequested (PFLAG_FORCE_LAYOUT in Android)
+			// could be reset without any call to onMeasure().
+
+			// The reason for that is unclear and it's only happening in very specific
+			// edge cases.
+
+			// It can be detected when the onDraw() is called before any call to onMeasure().
+
+			// This detection is done using the _measureRequested variable.
+
+			// GH Issue: https://github.com/unoplatform/uno/issues/6236
+
+			// Force apply the PFLAT_FORCE_LAYOUT on the control + parent hierarchy
+			super.requestLayout();
+
+			// Force the platform to requeue a layout pass
+			super.requestFitSystemWindows();
```

Comment:
It would probably be worth to add some (temporary) logging and see how often this path gets hit in a real app (eg an nventive app). Just to have some idea if there's a potential perf impact.

Reply:
@davidjohnoliver And how do you log something from java code?

Reply:
Logging is interesting, though it may log a lot. We should make it conditional to an environment variable (to avoid adding an API on UnoViewGroup).

Reply:
@carldebilly See eg:
https://github.com/unoplatform/uno/blob/b3f185f07e04726c90d5777aa3302bb4b0a03965/src/Uno.UI.BindingHelper.Android/Uno/UI/UnoViewGroup.java#L56-L64